sdk/Lusid.Sdk.Tests/tutorials/Instruments/CreditDefaultSwapCashFlow.cs

```diff
@@ -0,0 +1,134 @@
+            _aggregationApi = _apiFactory.Api<IAggregationApi>();
+            _testDataUtilities = new TestDataUtilities(
+                _apiFactory.Api<ITransactionPortfoliosApi>(),
+                _apiFactory.Api<IInstrumentsApi>(),
+                _apiFactory.Api<IQuotesApi>(),
+                _apiFactory.Api<IComplexMarketDataApi>());
+        }
+        
+        [Test]
+        public void ExamplePortfolioCashFlowsForCreditDefaultSwaps()
+        {
+            // CREATE portfolio
+            var portfolioScope = Guid.NewGuid().ToString();
+            var portfolioId = _testDataUtilities.CreateTransactionPortfolio(portfolioScope);
+
+            // CREATE CDS
+            var cds = InstrumentExamples.CreateExampleCreditDefaultSwap() as CreditDefaultSwap;
+        
+            // UPSERT CDS to portfolio and populating stores with required market data
+            var effectiveAt = new DateTimeOffset(2020, 2, 23, 0, 0, 0, TimeSpan.Zero);
+            _testDataUtilities.AddInstrumentsTransactionPortfolioAndPopulateRequiredMarketData(
+                portfolioScope, 
+                portfolioId,
+                effectiveAt,
+                effectiveAt,
+                cds);
+            
+            // UPSERT CDS spread curve before upserting recipe
+            _testDataUtilities.UpsertCdsSpreadCurves(portfolioScope, effectiveAt, cds.Ticker, cds.FlowConventions.Currency, cds.ProtectionDetailSpecification.Seniority, 
+                cds.ProtectionDetailSpecification.RestructuringType);
+            
+            // CREATE and upsert recipe specifying discount pricing model
+            var discountingRecipeCode = "DiscountingRecipe";
+            CreateAndUpsertRecipe(discountingRecipeCode, portfolioScope, ModelSelection.ModelEnum.Discounting);
+            
+            // CALL api to get cashflows at maturity
+            var maturity = cds.MaturityDate;
+            var cashFlowsAtMaturity = _transactionPortfoliosApi.GetPortfolioCashFlows(
+                portfolioScope,
+                portfolioId,
+                maturity.AddMilliseconds(-1),
+                maturity.AddMilliseconds(-1),
+                maturity.AddMilliseconds(1),
+                null,
+                null,
+                portfolioScope,
+                discountingRecipeCode);
+            
+            // CHECK correct number of cashflow at CDS maturity: There is 1 cash flows corresponding to the last coupon amount.
+            var expectedNumber = 1;
+            Assert.That(cashFlowsAtMaturity.Values.Count, Is.EqualTo(expectedNumber));
+            
+            var cashFlows = cashFlowsAtMaturity.Values.Select(cf => cf)
+                .Select(cf => (cf.PaymentDate, cf.Amount, cf.Currency))
+                .ToList();
+            
+            // CHECK that expected cash flows at maturity are not 0.
+            var allCashFlowsPositive = cashFlows.All(cf => cf.Amount > 0);
+            Assert.That(allCashFlowsPositive, Is.True);
```

Comment:
can we remove 
```
            var cashFlows = cashFlowsAtMaturity.Values.Select(cf => cf)
                .Select(cf => (cf.PaymentDate, cf.Amount, cf.Currency))
                .ToList();

            // CHECK that expected cash flows at maturity are not 0.
            var allCashFlowsPositive = cashFlows.All(cf => cf.Amount > 0);
            Assert.That(allCashFlowsPositive, Is.True);
```
this would change and break when protection leg goes in.

Same really with 
`Assert.That(cashFlowsAtMaturity.Values.Count, Is.EqualTo(expectedNumber));`.... maybe
`Assert.That(cashFlowsAtMaturity.Values.Count, Is.GreaterOrEqualTo(1))`;
but tbh, that can break too....

Reply:
@jz-finbourne why would this break when the protection leg gets created, you expect more cash flows to appear?

Reply:
I would assume/think so. Well, it is certainly possible. 
We don't want to check nor break that here. We test protection leg inside lusid.

Reply:
Do the cash flows returned carry some sort of a type? If we can look say for type "coupon" or "principal", perhaps we can test specifically for that cash flow here. 

And then if the protection leg gets added you'd be able to extend this.

Reply:
The test has now been updated and the cash flow is filtered as a premium leg payment.